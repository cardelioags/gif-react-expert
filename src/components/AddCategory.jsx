import { useState } from "react";

// eslint-disable-next-line react/prop-types
export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setinputValue] = useState('')

    const onInputChange = ({ target }) => {
        setinputValue(target.value);
    }
    const onSubmit = (event) => {
        event.preventDefault();
        const newInputValue = inputValue.trim();
        if (newInputValue.length <= 1) return;
        onNewCategory(newInputValue);
        setinputValue('');
    }
    return (
        <form onSubmit={onSubmit}>
            <input
                type='text'
                placeholder='Buscar Gifts'
                value={inputValue}
                onChange={onInputChange}
            ></input>
        </form>
    )
}