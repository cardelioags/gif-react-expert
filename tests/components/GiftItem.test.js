import { render } from "@testing-library/react";
import { GifItem } from '../../src/components/GifItem';

/* eslint-disable no-undef */
describe('Pruebas en <GifItem />', () => {

    const title = 'Batman';
    const url = 'https://batman.gif'
    test('Debe de hacer match con snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect( container ).toMatchSnapshot()
    })
})