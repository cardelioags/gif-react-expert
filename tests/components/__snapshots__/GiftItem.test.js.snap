// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Batman"
      src="https://batman.gif"
    />
    <p>
      Batman
    </p>
  </div>
</div>
`;
